src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.netcoreapp.cs

```diff
@@ -41,6 +41,20 @@ public static void CreateServer_ConnectClient()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)] // On Unix domain socket should have different location in this case.
```

Comment:
Interesting.  I admit I didn't yet review the CurrentUserOnly code that went in and so haven't thought deeply about it.  But it's a little worrying that CurrentUserOnly when applied on just the server or client means that the current user can't connect to it.  Do we have other approaches we could take that would enable it?  Could we get away with only checking user ID or we believe the extra security provided by the directory is critical?  Could we try one path and then the other if we fail to connect on the first?  Etc.

Reply:
We could get away with removing the extra directory to avoid this behavior, the downside seems that we need throw on the server side, see https://github.com/dotnet/corefx/pull/26395#discussion_r166484692

There is also a small incorrect thing that I just noticed about the current code: we chmod the directory but don't check actual ownership so for instance root can chmode but owner is still old user (the code ignores that for now).